corehq/motech/repeaters/models.py

```diff
@@ -174,6 +183,44 @@ def log_repeater_success_in_datadog(domain, status_code, repeater_type):
     })
 
 
+class RepeaterSuperProxy(models.Model):
+    # See https://stackoverflow.com/questions/241250/single-table-inheritance-in-django/60894618#60894618
+    PROXY_FIELD_NAME = "repeater_type"
+
+    repeater_type = models.CharField(max_length=64, blank=True)
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        self.repeater_type = self._repeater_type
+        return super().save(*args, **kwargs)
+
+    def __new__(cls, *args, **kwargs):
+        repeater_class = cls
+        try:
+            # get proxy name, either from kwargs or from args
+            proxy_class_name = kwargs.get(cls.PROXY_FIELD_NAME)
+            if proxy_class_name is None:
+                proxy_name_field_index = cls._meta.fields.index(
+                    cls._meta.get_field(cls.PROXY_FIELD_NAME))
+                proxy_class_name = args[proxy_name_field_index]
+            # Since we are storing original repeater names so we need to append SQL
+            # This should be removed when SQL is dropped from model names
+            proxy_sql_class_name = f"SQL{proxy_class_name}"
+            # get proxy class, by name, from current module
+            repeater_class = getattr(sys.modules[__name__], proxy_sql_class_name)
+        except IndexError:
+            pass
+        finally:
+            return super().__new__(repeater_class)
```

Comment:
I think the finally should be removed. Otherwise it will silently swallow errors (other than `IndexError`, which is explicitly silenced).
```suggestion
        return super().__new__(repeater_class)
```